MiniAn/minian_main.py

```diff
@@ -467,6 +472,119 @@ def load_doric_to_xarray(
+    cents = calculate_centroids(A_shifted, window)
+
+    # Calculate pairwise distance between cells in all pairs of sessions. 
+    # Note that at this stage, since we are computing something along the session dimension, 
+    # it is no longer considered as a metadata dimension, so we remove it    
+    id_dims =  []
+    dist = calculate_centroid_distance(cents, "session", id_dims)
+
+    # Threshold centroid distances, keeping only cell pairs with distance less than param_dist.
+    param_dist = 5
+    dist_ft = dist[dist["variable", "distance"] < param_dist].copy()
+    dist_ft = group_by_session(dist_ft)
+
+    # Generate mappings
+    mappings = calculate_mapping(dist_ft)
+    mappings_meta = resolve_mapping(mappings)
+    mappings_meta_fill = fill_mapping(mappings_meta, cents)
+    mappings_meta_fill.head()
+
+    # Update unit ids of the current file if they have a mapped unit id in ref file
+    """
+    Psedocode:
+    1. Create an empty list (with zeros) of the same size as A['unit_ids']
+       This list will be modifiled/updated according to the mapping in 2 sessions.
+    2. Create another list (copy of A[unit_ids]) to be used as a reference to check the index of the values (list_unit_ids)
+    3. Loop through the mapping output (mapping_meta_fill) 
+        a. If a cell is in both sessions-
+           Get index of that cell from list_unit_ids and update the value  in updated_list to id in ref session
+        b. If a cell is only in current session- Update the value to unitId_max 
+    """
+    unitId_max       = A_ref.coords["unit_id"].values.max() + 1
+    list_unit_ids    = list(A["unit_id"].values)
+    updated_unit_ids = [0]*len(list_unit_ids)
+
+    for i in range(len(mappings_meta_fill)):
+        if mappings_meta_fill.iloc[i]["group"][0] == ("session1", "session2"):
+            index = list_unit_ids.index(mappings_meta_fill.iloc[i]["session"]["session2"])
+            updated_unit_ids[index] = int(mappings_meta_fill.iloc[i]["session"]["session1"])
+        elif mappings_meta_fill.iloc[i]["group"][0] == ("session2",):
+            index = list_unit_ids.index(mappings_meta_fill.iloc[i]["session"]["session2"])
+            updated_unit_ids[index] = int(unitId_max)
+            unitId_max += 1
+
+    A["unit_id"] = updated_unit_ids
+    return A
+
+
+def get_footprints(filename, rois_h5path, dims):
+    
+    with h5py.File(filename, 'r') as file_:
+
+        roi_names =  list(file_.get(rois_h5path))
+
+        roi_ids = np.zeros((len(roi_names) - 1))
+        footprints = np.zeros(((len(roi_names) - 1), dims["height"].size, dims["width"].size), np.float32)
+
+        for i in range(len(roi_names) - 1):
+            attrs = utils.load_attributes(file_, f"{rois_h5path}/{roi_names[i]}")
+
+            roi_ids[i] = int(attrs["ID"])
+
+            coords = np.array(attrs["Coordinates"])
+            mask = np.zeros((dims["height"].size, dims["width"].size), np.float32)
```

Comment:
Has to be `np.float32`?

Reply:
checked. not necessarily float32. so changed it to 64

Reply:
I would explicitly to `np.float64` to avoid confusions